Sources/Testing/Test.swift

```diff
@@ -192,29 +192,18 @@ public struct Test: Sendable {
     containingTypeInfo != nil && testCasesState == nil
   }
 
-  /// Whether or not this instance was synthesized at runtime.
-  ///
-  /// During test planning, suites that are not explicitly marked with the
-  /// `@Suite` attribute are synthesized from available type information before
-  /// being added to the plan. For such suites, the value of this property is
-  /// `true`.
-  @_spi(ForToolsIntegrationOnly)
-  public var isSynthesized = false
```

Comment:
This property is useful for diagnosing issues with synthesized instances—why did you remove it?

Reply:
It doesn't appear to be used anywhere

Reply:
It's not actively used but it's _useful_ when I'm debugging stuff.

Reply:
What if we made `isSynthesized` into an `internal`-only, non-recursive `SuiteTrait` instead? Then we could eliminate this stored property, while keeping the ability to see whether a suite is synthesized while debugging.

And if we take that approach, I would like to conditionalize the trait's inclusion using `#if DEBUG`.

Reply:
For this PR, I have restored the `isSynthesized` property, and actually adopted it in my new unit tests so it's no longer unused